src/export/StructureDefinitionExporter.ts

```diff
@@ -414,9 +415,12 @@ export class StructureDefinitionExporter implements Fishable {
         // The FHIR spec prohibits constraining any parent element in a 'specialization'
         // (i.e., logical model and resource), therefore log an error if that is attempted
         // and continue to the next rule.
-        const baseElementPath = element.path.replace(`${element.structDef.id}.`, '');
-        const addElementRules = fshDefinition.rules.filter(rule => rule instanceof AddElementRule);
-        if (rule.path && !addElementRules.some(rule => baseElementPath.includes(rule.path))) {
+        if (
+          rule.path &&
+          !addElementRules.some(rule =>
+            element.path.startsWith(`${element.structDef.pathType}.${rule.path}`)
```

Comment:
This isn't quite right.  Consider:
* elementPath: `'Foo.bars'`
* rule.path: `'bar'`
Then the above line is equivalent to:
```ts
'Foo.bars'.startsWith('Foo.bar')
```
which is true, but that's not the intent.  I think you want:
```ts
element.path === `${element.structDef.pathType}.${rule.path}` ||
element.path.startsWith(`${element.structDef.pathType}.${rule.path}.`)
```
which, in our example would be equivalent to:
```ts
'Foo.bars' === 'Foo.bar' || 'Foo.bars'.startsWith('Foo.bar.')
```
That is false, which is what I think we intend since there is no subpath relationship here.

Reply:
This is totally right, thank you for finding this!